src/libraries/System.Private.CoreLib/src/System/Text/CompositeFormat.cs

```diff
@@ -332,6 +339,21 @@ private static bool TryParseLiterals(ReadOnlySpan<char> format, List<(string? Li
                 // Continue parsing the rest of the format string.
             }
 
+            FailureUnexpectedClosingBrace:
```

Comment:
I'm probably misremembering. I was under the impression that choosing your resource in advance instead of immediately throwing with it was kind of expensive.